sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/BodyMemory.cs

```diff
@@ -6,96 +6,144 @@
 using System.Collections.Generic;
```

Comment:
@JoshLove-msft:  Any thoughts on this fitting into `Azure.Core.Amqp` as shared source?   This doesn't seem bound to Service Bus logic and we're likely to want to use this in Event Hubs as well.

Reply:
Not in it current form at least as it depends on Microsoft.Azure.Amqp. I think we can look into factoring it out when we get to the Event Hubs implementation.

Reply:
Fair;  I forgot about that.